Scripts

TF20083540 opened the pull request:
Requesting to merge in current work on the manager. That way I can pull the most recent build with everything else and not lose what I've got too.

TF20083540 committed:
Manager movement (Broken)

Manager makes it to one place and wobbles (precision issue?), walks far past the second point and also wobbles.

Tried multiple methods to fix the issue, still working on it.

TF20083540 committed:
Update s_Manager_NPC.cs

Manager now moving and stopping as expected. Will be working on flipping his sprite and working on detecting the player doing something they shouldn't be next.

TF20083540 committed:
Merge branch 'master' of https://github.com/Xwilarg/MiniJameGam16 into Scripts

TF20083540 committed:
Manager NPC Code Finished

As far as I can see, I've accounted for everything that should be needed. A reference to the player for punishment, a way to walk back and forth, a way to know if they can actually punish the player, a cooldown on punishment. Everything SHOULD be done.

Xwilarg committed:
Merge branch 'master' into Scripts

Xwilarg commented on Assets/Scripts/NPCs/s_Manager_NPC.cs:
```diff
@@ -0,0 +1,151 @@
+        }
+
+        //Real detection code.
+        if(hits.Length > 0) //If the player is detected, they are in view, and may now be punished if they hit the Duct Tape Button.
```
```suggestion
        if (hits.Any()) //If the player is detected, they are in view, and may now be punished if they hit the Duct Tape Button.
```

TF20083540 replied:
Now fixed.

Xwilarg commented on Assets/Scripts/NPCs/s_Manager_NPC.cs:
```diff
@@ -0,0 +1,151 @@
+        if(hits.Length > 0) //If the player is detected, they are in view, and may now be punished if they hit the Duct Tape Button.
+        {
+
+            if(Input.GetKeyDown(KeyCode.Space) && punishmentEnabled == true){ //Replace "Input.GetKeyDown(KeyCode.Space)" with a call to the player controller.
```
This code is using the old input system

TF20083540 replied:
yes, "Input.GetKeyDown(KeyCode.Space" will instead be a reference to the player controller, so that the script can detect if that button has been pressed. So that will change once I've got the real player controller.

Xwilarg commented on Assets/Scripts/NPCs/DummyPlayer.cs:
```diff
@@ -0,0 +1,30 @@
+using System.Collections;
+using System.Collections.Generic;
```
```suggestion
```

TF20083540 replied:
Now fixed.

Xwilarg commented on Assets/Scripts/NPCs/DummyPlayer.cs:
```diff
@@ -0,0 +1,30 @@
+
+public class DummyPlayer : MonoBehaviour
+{
+    // Start is called before the first frame update
+    void Start()
+    {
+        
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
```
```suggestion
```

TF20083540 replied:
DummyPlayer is simply a class I used for checking some things. It won't be used in the final product.

Xwilarg commented on Assets/Scripts/NPCs/s_Manager_NPC.cs:
```diff
@@ -0,0 +1,151 @@
+            //Branching action based on the time until it's allowed to walk again.
+            if (currentWaitTime != 0)  //If the time is not zero.
+            {
+                currentWaitTime--;  //Decrement the time by 1.
```
Timer is framerate dependant

Xwilarg commented on Assets/Scripts/NPCs/s_Manager_NPC.cs:
```diff
@@ -0,0 +1,151 @@
+
+            Debug.Log("Hits size: " +hits.Length);  //Just checking if it was a per pixel thing. It's per object.
+        }
+        else //If the player is not detected, do nothing.
+        {
+
+        }
```
```suggestion
```

TF20083540 committed:
Fixes to Manager NPC script.

Multiple changes made as requested to two files. However, the old input code still needs to be replaced. Still looking into that as I've not worked with the new system yet.

TF20083540 committed:
Merge branch 'Scripts' of https://github.com/Xwilarg/MiniJameGam16 into Scripts

TF20083540 committed:
Input system changed.

Changed from the old system to the new system. Warning, InputAction "DuckTape" required in PlayerInput, else causes a null reference exception. Whoever is working on the action mappings, let me know the name of the actual "DuckTape" button and it should be fixed after a merge.

That is the only problem this code faces.

So basically fix up the PlayerController & PlayerInput stuff and it should be good to go.
